### PR TITLE
Fix (Boilerplate): Add Event Parameter to test, Remove Infosec Requriements, Add Lowercase Requirement

### LINKED ISSUE #39616: Imperial Metric Converter - Boilerplate/Repo/Repl.it "Convert" button does not work
**Describe your problem and how to reproduce it:**
If you click the "Convert" button on the UI for the example project/boilerplate/Repl.it, the page refreshes and the conversion does not occur.  This is because the `onsubmit` handler in the script does not have an `event` parameter being passed in, despite doing the `event.preventDefault()` in the body. Adding the `event` parameter fixes the issue.

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/quality-assurance/quality-assurance-projects/metric-imperial-converter
https://metric-imperial-converter--freecodecamp.repl.co/
https://github.com/freeCodeCamp/boilerplate-project-metricimpconverter/

I have a fix here (amongst other changes):
https://github.com/freeCodeCamp/boilerplate-project-metricimpconverter/pull/12

### LINKED ISSUE #39618: Imperial/Metric Conversion - Boilerplate/Repl.it has Incorrect Example conversion
**Describe your problem and how to reproduce it:**
The `index.html` has an example conversion output:
>`<code>{initNum: 3.1, initUnit: 'mi', returnNum: 5.0000008, returnUnit: 'km', string: '3.1 miles converts to 5.00002 kilometers'}</code>`

As per Google and basic math:
3.1mi * 1.60934 km/mi = 4.98895 not 5.0000008
The results are also supposed to be rounded to 5 digits.

This appears to be correct in the sample project (or at least within 2 hundred thousandths)

**Add a Link to the page with the problem:**
https://github.com/freeCodeCamp/boilerplate-project-metricimpconverter/blob/gomix/views/index.html#L42

I have a fix here:
https://github.com/freeCodeCamp/boilerplate-project-metricimpconverter/pull/12

### PR DESCRIPTION
**Pull request by SaintPeter**
Resolves several issues with the boilerplate code. I have tested all of these changes locally.

* Add an `event` parameter to the `submit` event handler in `index.html` so that the commit button will work properly.  Currently it submits the page because `preventDefault()` cannot execute without an `event` parameter.
* Remove the Infosec requirements, as per freeCodeCamp/freeCodeCamp#39435
* Add a "units to lowercase" requirement.  This is consistent with the current test cases, reduces ambiguity, and improves standardization.
* Corrects the "example conversion" numbers which were incorrect, as per Google and basic math:
3.1mi * 1.60934 km/mi = 4.98895 not 5.0000008
The results are also supposed to be rounded to 5 digits.

I've made additional changes:
* Fixed the `assertion-analyzer.js` as I have on other projects
* Updated packages to latest
* Move to let/const declarations
* Implemented the "lowercase units except for liters" change
* Added color for `<code>` tags background.

Closes #39617 - Inconsistency in unit capitalization

Closes https://github.com/freeCodeCamp/freeCodeCamp/issues/39618
Closes https://github.com/freeCodeCamp/freeCodeCamp/issues/39616

### CODE REVIEW

File: tests/2_functional-tests.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ suite('Functional Tests', function() {
         .end(function(err, res){
           assert.equal(res.status, 200);
           assert.equal(res.body.initNum, 10);
-          assert.equal(res.body.initUnit, 'L');
+          assert.equal(res.body.initUnit, 'l');
```
Comment (ShaunSHamilton) [inline tests/2_functional-tests.js:29]:
```suggestion
          assert.equal(res.body.initUnit, 'L');
```

File: views/index.html [hunk 2]
```diff
@@ -18,13 +18,12 @@ <h1>
...
       <ol>
-        <li>I will help prevent the client from trying to guess(sniff) the MIME type.</li>
-        <li>I will prevent cross-site scripting (XSS) attacks.</li>
         <li>I can <b>GET</b> <code>/api/convert</code> with a single parameter containing an accepted number and unit and have it converted.</li>
         <li>Hint: Split the input by looking for the index of the first character.</li>
-        <li>I can convert 'gal' to 'L' and vice versa. <b>(1 gal to 3.78541 L)</b></li>
+        <li>I can convert 'gal' to 'l' and vice versa. <b>(1 gal to 3.78541 l)</b></li>
         <li>I can convert 'lbs' to 'kg' and vice versa. <b>(1 lbs to 0.453592 kg)</b></li> 
         <li>I can convert 'mi' to 'km' and vice versa. <b>(1 mi to 1.60934 km)</b></li>
+        <li>All units must be returned lowercase.</li>
```
Comment (ShaunSHamilton) [inline views/index.html]: Why are you adding this condition?

File: views/index.html [hunk 3]
```diff
@@ -18,13 +18,12 @@ <h1>
     <div id='userstories' style='margin-left: 5%; margin-top: 5%'>
       <h3>User Stories</h3>
       <ol>
-        <li>I will help prevent the client from trying to guess(sniff) the MIME type.</li>
-        <li>I will prevent cross-site scripting (XSS) attacks.</li>
         <li>I can <b>GET</b> <code>/api/convert</code> with a single parameter containing an accepted number and unit and have it converted.</li>
         <li>Hint: Split the input by looking for the index of the first character.</li>
-        <li>I can convert 'gal' to 'L' and vice versa. <b>(1 gal to 3.78541 L)</b></li>
+        <li>I can convert 'gal' to 'l' and vice versa. <b>(1 gal to 3.78541 l)</b></li>
```
Comment (ShaunSHamilton) [inline views/index.html]:
```suggestion
        <li>I can convert 'gal' to 'L' and vice versa. <b>(1 gal to 3.78541 L)</b></li>
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
These are not the changes you mentioned on Discord. So, I would suggest an issue be opened on the main repo, if any more are to be made.

I do not see a reason for the `L` unit to be mentioned or returned in lowercase.

**Comment by SaintPeter** [comment]
I'll open an issue, but here is the issue:
How do you handle capitalization in a standardized manner?
The example tests show both all caps and all lowercase.  If you're passed in a capital L should you return that?  What about the return units?

I use an object to keep track of the conversions and to determine if a unit is convertible.  The easiest thing to do is take everything to lowercase so it's consistent.  

Anyway, I'll open up an issue in the main repo and reference this.

**Comment by SaintPeter** [comment]
Ref: freeCodeCamp/freeCodeCamp#39616 for `event` issue.

**Comment by ShaunSHamilton** [comment]
> How do you handle capitalization in a standardized manner?
The example tests show both all caps and all lowercase. If you're passed in a capital L should you return that? What about the return units?

I would say the app should be able to have both caps and lowercase unit inputs. So, to standardise: what is put in, is put out.
If this is a point of confusion, I suggest we add a note about returning what is given

To avoid bikeshedding, let us hear from another contributor as to whether we should expect the output to always be lowercase, or to be what is put in. @scissorsneedfoodtoo Do you have an opinion on this?

>  The easiest thing to do is take everything to lowercase so it's consistent.

I do the same, but only for comparison, then return what is given.

EDIT: Thanks for opening up those issues. I think one would have done, though 😄

**Comment by SaintPeter** [comment]
> EDIT: Thanks for opening up those issues. I think one would have done, though

Hey, buddy, you want issues, you get issues.  ;)
I'm still getting back on the bike here, it's been ~5 years since I was a regular contributor.  Now, once again, I'm going through the curriculum and finding issues that I just can't ignore.

**Comment by scissorsneedfoodtoo** [comment]
@SaintPeter, thanks for opening this PR and for creating those issues. It really helps us keep track of the changes that need to be made. Also, thank you for your review @Sky020.

As for the letter case of the input, I generally think that the app should be able to handle both upper and lowercase inputs. We do want to be more explicit with the expected output for this and the other backend projects, though.

For the return units, I could honestly go either way. It seems like the only one that's regularly capitalized is L, so it would be easier to only return lowercase units. But if we decide to return upper or lowercase based on the input, that would also be an interesting requirement. Either way, we would need a note or another user story explaining this.

@moT01 and @RandellDawson, what do you both think?

**Comment by SaintPeter** [comment]
I've made additional changes:
* Fixed the `assertion-analyzer.js` as I have on other projects
* Updated packages to latest
* Move to let/const declarations
* Implemented the "lowercase units except for liters" change
* Added color for `<code>` tags background.

Closes #39617 - Inconsistency in unit capitalization

**Comment by SaintPeter** [comment]
I made a few more cosmetic changes to make the user story verbiage match the site tests from that associated PR.

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍 

Although, why did you include MongoDB in the packages?

**Comment by naomi-lgbt** [comment]
> Although, why did you include MongoDB in the packages?

It looks like MongoDB was already [included in the packages](https://github.com/freeCodeCamp/boilerplate-project-metricimpconverter/blob/gomix/package.json#L17)?

Edit: I might have misunderstood the question - MongoDB could likely be removed from this boilerplate.

**Review by naomi-lgbt — APPROVED** [review]
Changes look good to me. Thanks @SaintPeter~!

This will need to wait for merge until [#39615](https://github.com/freeCodeCamp/freeCodeCamp/pull/39615) hits production, I believe.

**Comment by SaintPeter** [comment]
@Sky020 Since we're waiting, I yeeted the mongdb ref.

File: README.md [hunk 4]
```diff
@@ -1,9 +1 @@
-**FreeCodeCamp**- Information Security and Quality Assurance
-------
-[![Run on Repl.it](https://repl.it/badge/github/freeCodeCamp/boilerplate-project-metricimpconverter)](https://repl.it/github/freeCodeCamp/boilerplate-project-metricimpconverter)
-1) SET NODE_ENV to `test` without quotes
-2) Most logic will need done in `controllers/convertHandler.js` but do complete `routes/api.js`
-3) You will add any security features to `server.js`
-4) You will create all of the functional/unit tests in `tests/2_functional-tests.js` and `tests/1_unit-tests.js`
-
-
+# [Quality Assurance Projects - Metric-Imperial Converter](https://www.freecodecamp.org/learn/quality-assurance/quality-assurance-projects/metric-imperial-converter)
```
Comment (ShaunSHamilton) [inline README.md]:
```suggestion
# [Metric-Imperial Converter](https://www.freecodecamp.org/learn/quality-assurance/quality-assurance-projects/metric-imperial-converter)
```

File: views/index.html [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>Welcome to HyperDev!</title>
+    <title>Quality Assurance Projects - Metric/Imperial Converter</title>
```
Comment (ShaunSHamilton) [inline views/index.html]:
```suggestion
    <title>Metric/Imperial Converter</title>
```

File: views/index.html [hunk 6]
```diff
@@ -12,36 +12,9 @@
   <body>
     <header>
       <h1>
-        ISQA_2 - Metric/Imp Converter
+        Quality Assurance Projects - Metric/Imperial Converter
```
Comment (ShaunSHamilton) [inline views/index.html]:
```suggestion
        Metric/Imperial Converter
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
In keeping with the previously mentioned changes

File: views/index.html [hunk 7]
```diff
@@ -12,36 +12,9 @@
...
-        <li>All 5 functional tests are complete and passing.</li>
-      </ol>
-      <h3>Example usage:</h3>
-      <code>/api/convert?input=4gal</code><br>
-      <code>/api/convert?input=1/2km</code><br>
-      <code>/api/convert?input=5.4/3lbs</code><br>
-      <code>/api/convert?input=kg</code><br>
-      <h3>Example return:</h3>
-      <code>{initNum: 3.1, initUnit: 'mi', returnNum: 5.0000008, returnUnit: 'km', string: '3.1 miles converts to 5.00002 kilometers'}</code>
-    </div>
```
Comment (moT01) [inline views/index.html]: I think we may be keeping the example usages on the boilerplates. That's what I've been doing anyway, we may need to discuss it on the issue.

**Comment by SaintPeter** [comment]
@moT01 I took a stab at reformatting the `index.html`, adding back in the example text:
image

**Review by RandellDawson — APPROVED** [review]
@SaintPeter I am happy to merge this.  Someone still needs to make necessary changes mentioned in https://github.com/freeCodeCamp/freeCodeCamp/issues/39721, but those changes should be in a separate PR.

Thanks for all your hard work on this.

We will need to wait until the `/learn` PR and corresponding `demo-projects` PR hit production before we can merge this.

**Comment by SaintPeter** [comment]
@RandellDawson I do have a PR open for those changes:
https://github.com/freeCodeCamp/freeCodeCamp/pull/39615/files

It just needs to be merged.  Unless it needs additional instructions?  If so, please let me know and I'll make a change to that PR.

**Comment by RandellDawson** [comment]
@SaintPeter I updated the `/learn` PR.  Sorry for the confusion.  This boilerplate PR is fine, but we need the example project that we link to on `/learn` to pass all the tests before we can merge the `/learn` PR.  I have explained everything on the `/learn` PR.

**Comment by RandellDawson** [comment]
@SaintPeter Thanks for adding those last two commits.  That cleans up things.

@Sky020 I am happy will all the changes.  If you are, then go ahead and approve, so we can merge once the `/learn` changes hit production.

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**